### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException: EventRecurrenceFormatter.java:110

### ISSUE THREAD

**Issue by mtrakal** [code-troopers/android-betterpickers#282]
Latest 2.5.3 version:

For some users it crash on this after update lib. It's from crashlytics, so I'm sorry I doesn't have more info :/

```
Fatal Exception: java.lang.ArrayIndexOutOfBoundsException: length=5; index=-2
       at com.codetroopers.betterpickers.recurrencepicker.EventRecurrenceFormatter.getRepeatString(EventRecurrenceFormatter.java:110)
```

**Comment by fchauveau** [comment]
Hi @mtrakal, 
Can you have language of user? This may help.

**Comment by mtrakal** [comment]
Looks like Spain with es, English (Switzerland with en, USA with en). We are using more translations (to support more langs), but every looks OK. Has same rows (`5`) as it is said in exception.

But it looks on problem with index `-2`, not with arrays in strings.

Ok, I try to reproduce and it is by set value:`
bundle.putLong(RecurrencePickerDialogFragment.BUNDLE_START_TIME_MILLIS, LAST_SUNDAY_OF_MONTH.getMillis());`

click on Monthly, select: every last sunday.

When I click OK, code go to: `EventRecurrenceFormatter.java:110` and dayNumber is set to `-1`.
So it try to get: `mMonthRepeatByDayOfWeekStrs[6][-2];`

recurrence.bydayNum[0] is -1, but should be: 4, for my condition.

It's independant on lang, it depends on selected start day

**Comment by mtrakal** [comment]
![clipboard01](https://cloud.githubusercontent.com/assets/1458891/15328868/cf19e630-1c56-11e6-909c-3fbf3a8e9aee.png)

**Comment by mtrakal** [comment]
![clipboard02](https://cloud.githubusercontent.com/assets/1458891/15328935/2b5b9d8a-1c57-11e6-89f7-8e9a5de3202d.png)

**Comment by fchauveau** [comment]
Hey @mtrakal 
Can you have give a try to https://github.com/code-troopers/android-betterpickers/tree/fix/282
I think it's correctly fix this issue, but if you can try it before i release it.

Thx

**Comment by mtrakal** [comment]
Looks fixed :+1:

**Comment by fchauveau** [comment]
fixed in 2.5.5, release in progress
